pkg/controller/pravegacluster/upgrade.go

```diff
@@ -322,10 +322,82 @@ func (r *ReconcilePravegaCluster) syncControllerVersion(p *pravegav1alpha1.Prave
 	return true, nil
 }
 
+func (r *ReconcilePravegaCluster) syncStoreVersion(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	//this is to check if the target image is above 07 and the current image is below 07
+	if util.IsClusterUpgradingTo07(p) {
+		return r.syncSegmentStoreVersionTo07(p)
+	}
+	return r.syncSegmentStoreVersion(p)
+}
+
+func (r *ReconcilePravegaCluster) syncSegmentStoreVersionTo07(p *pravegav1alpha1.PravegaCluster) (synced bool, err error) {
+	newsts := pravega.MakeSegmentStoreStatefulSet(p)
+	controllerutil.SetControllerReference(p, newsts, r.scheme)
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: newsts.Name, Namespace: p.Namespace}, newsts)
+	//this check is to see if the newsts is present or not if it's not present it will be created here
+	if err != nil {
```

Comment:
to check if the STS exists or not, you should be checking using:
``` if err != nil  && errors.IsNotFound(err) ```

Reply:
done